atlp-rwanda/e-commerce-lydia-32-fn
# Implementing User Wishlist feature

### What does this PR do?
This PR implements the user wishlist feature on the frontend, allowing users to 
View, manage, and delete items from their wishlist.

### Description of Task to be completed?

1. Fetch wishlist items for the logged-in user from the backend.
2. Display the fetched wishlist items in a user-friendly manner.
3. Add functionality to delete items from the wishlist with confirmation.
4. Ensure the wishlist data is updated in the Redux store and the UI is refreshed accordingly.
5. Handle loading states and errors during data fetching and item deletion.

### How should this be manually tested?

1. Log in to the application.
2. Navigate to the wishlist page.
3. Verify that the wishlist items are displayed correctly.
4. Attempt to delete an item and confirm the action.
5. Ensure that the item is removed from the wishlist and the UI is updated.
6. Check for proper handling of loading states and error messages.


#### N/A

### Any background context you want to provide?

#### N/A

### Screenshots (if appropriate)

#### N/A

### Questions:

#### N/A

### What are the relevant pivotal tracker stories
#187570838

#### N/A
